controller/array_action/array_mediator_svc.py

```diff
@@ -468,26 +474,17 @@ def _stop_and_delete_fcmap(self, fcmap_id):
         self._stop_fcmap(fcmap_id)
         self._delete_fcmap(fcmap_id, force=True)
 
-    def _delete_all_fcmaps_as_source_if_exist(self, snapshot_name):
-        snapshot_as_source_fcmaps = self._get_fcmaps(snapshot_name, ENDPOINT_TYPE_SOURCE)
-        unfinished_fcmaps = [fcmap.name for fcmap in snapshot_as_source_fcmaps
-                             if fcmap.status != FCMAP_STATUS_DONE]
-        if unfinished_fcmaps:
-            raise controller_errors.ObjectIsStillInUseError(snapshot_name, unfinished_fcmaps)
-        for fcmap in snapshot_as_source_fcmaps:
-            self._delete_fcmap(fcmap.id, force=False)
-
-    def _delete_snapshot(self, cli_volume):
-        snapshot_name = cli_volume.name
-        fcmap = self._get_fcmap_as_target_if_exists(snapshot_name)
-        if not fcmap:
-            raise controller_errors.ObjectNotFoundError(snapshot_name)
-
-        fcmap_id = cli_volume.FC_id
-        if fcmap_id == 'many':
-            self._delete_all_fcmaps_as_source_if_exist(snapshot_name)
-        self._stop_and_delete_fcmap(fcmap.id)
-        self._delete_volume_by_name(snapshot_name)
+    def _delete_object(self, cli_object, snapshot=False):
+        object_name = cli_object.name
+        fcmap_as_target = self._get_fcmap_as_target_if_exists(object_name)
+        if snapshot and not fcmap_as_target:
+            raise controller_errors.ObjectNotFoundError(object_name)
+        if cli_object.FC_id == 'many':
+            fcmaps_as_source = self._get_fcmaps_as_source_if_exist(object_name)
+            self._safe_delete_fcmaps_as_source(fcmaps_as_source)
```

Comment:
if `_safe_delete_fcmaps_as_source` always requires a prior call to `_get_fcmaps_as_source_if_exist`, why not include the call inside it (as in the original `_delete_all_fcmaps_as_source_if_exist`)?

Reply:
done